src/test/java/mate/academy/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,169 @@
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.Set;
+import mate.academy.bookstore.dto.book.BookDto;
+import mate.academy.bookstore.dto.book.BookSearchParametersDto;
+import mate.academy.bookstore.dto.book.CreateBookRequestDto;
+import mate.academy.bookstore.service.BookService;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    private static final String ID = "/{id}";
+    private static final String TITLE_PARAM_NAME = "title";
+    private static final String AUTHOR_PARAM_NAME = "author";
+    private static final String ISBN_PARAM_NAME = "isbn";
+    private static final String PRICE_PARAM_NAME = "price";
+    private static final String BASE_URL = "/books";
+    private static final String SEARCH_URL = BASE_URL + "/search";
+    private static final String ADMIN_ROLE = "ADMIN";
+    private static final String USER_ROLE = "USER";
+    private static final String VALID_BOOK_TITLE = "Kobzar";
+    private static final String VALID_BOOK_AUTHOR = "Taras Shevchenko";
+    private static final String VALID_BOOK_ISBN = "978-0-7847-5628-7";
+    private static final BigDecimal VALID_BOOK_PRICE = BigDecimal.valueOf(14.99);
+    private static final Set<Long> VALID_CATEGORY_ID = Set.of(1L);
+    private static final Long VALID_BOOK_ID = 1L;
+    private static final String TITLE_0_EXPRESSION = "$[0].title";
+    private static final String TITLE_EXPRESSION = "$.title";
+
+    private static MockMvc mockMvc;
+
+    @MockBean
+    private BookService bookService;
```

Comment:
I think our api interation tests should not use mocked services, but rather a real service and a testcontainer DB. Same for the category controller test